Ports/qt6-qt5compat/package.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='qt6-qt5compat'
+version='6.2.3'
+workdir="qt5compat-everywhere-src-${version}"
+useconfigure='true'
+files="https://download.qt.io/official_releases/qt/$(cut -d. -f1,2 <<< ${version})/${version}/submodules/qt5compat-everywhere-src-${version}.tar.xz qt6-qt5compat-${version}.tar.xz 1cf89198cf2cf8a5c15336ccd69fa1f39b779feb64117d6bbf5509c21c123f53"
+auth_type='sha256'
+configopts=(
+    '-GNinja'
+    "-DCMAKE_TOOLCHAIN_FILE=${SERENITY_BUILD_DIR}/CMakeToolchain.txt"
+    '-DCMAKE_CROSSCOMPILING=ON'
+    '-DQT_BUILD_TOOLS_WHEN_CROSSCOMPILING=ON'
+    '-DQT_HOST_PATH=/usr'
+    '-DQT_FEATURE_cxx20=ON'
+)
+depends=(
+    'qt6-qtbase'
+    'libiconv'
+)
+
+QT_HOST_PATH='/usr'
```

Comment:
Not sure how I feel about hardcoding this.

Reply:
I've taken most of this from qt6-qtbase. The hard-coded values don't even work for me, since on Debian/Ubuntu this lives under `/usr/lib/x86_64-linux-gnu/cmake/`, but I tried to be consistent with qt6-qtbase. How should I change this?

Reply:
It looks like `qtpaths` / `qtpaths-qt5` exists in my system and points to the correct installation directory when called with `--install-prefix` as an argument. Could you check whether that is the case for your setup as well?

Reply:
Not quite, my versions of `qtpaths` apparently don't have a `--query` option at all.

Reply:
I now use qmake for that. I hope it works for all distros...

Reply:
fwiw I installed qt6 through their launcher rather than a package manager and the `qmake6` approach points to the right place and I don't even have `qtpaths` installed.

we should query qmake for the host path as well, since uh... macOS exists :)

```
qmake6 -query
QT_SYSROOT:
QT_INSTALL_PREFIX:/opt/homebrew
QT_INSTALL_ARCHDATA:/opt/homebrew/share/qt
QT_INSTALL_DATA:/opt/homebrew/share/qt
QT_INSTALL_DOCS:/opt/homebrew/share/qt/doc
QT_INSTALL_HEADERS:/opt/homebrew/include
QT_INSTALL_LIBS:/opt/homebrew/lib
QT_INSTALL_LIBEXECS:/opt/homebrew/share/qt/libexec
QT_INSTALL_BINS:/opt/homebrew/bin
QT_INSTALL_TESTS:/opt/homebrew/share/qt/tests
QT_INSTALL_PLUGINS:/opt/homebrew/share/qt/plugins
QT_INSTALL_QML:/opt/homebrew/share/qt/qml
QT_INSTALL_TRANSLATIONS:/opt/homebrew/share/qt/translations
QT_INSTALL_CONFIGURATION:
QT_INSTALL_EXAMPLES:/opt/homebrew/share/qt/examples
QT_INSTALL_DEMOS:/opt/homebrew/share/qt/examples
QT_HOST_PREFIX:/opt/homebrew
QT_HOST_DATA:/opt/homebrew/share/qt
QT_HOST_BINS:/opt/homebrew/bin
QT_HOST_LIBEXECS:/opt/homebrew/share/qt/libexec
QT_HOST_LIBS:/opt/homebrew/lib
QMAKE_SPEC:macx-clang
QMAKE_XSPEC:macx-clang
QMAKE_VERSION:3.1
QT_VERSION:6.4.0
```

Reply:
Thank you very much! I hope `QT_HOST_PREFIX` is also correct in your case?